### ISSUE TITLE
[feature request] Support setting environment in `local()` and `local_resource()`

### ISSUE THREAD

**Issue by KarolisL** [tilt-dev/tilt#3767]
I've got to execute some local binary with complex environment which should be generated dynamically.

At the moment I do something like this:
```python
def with_env(cmd, env):
    variables = " ".join(["{k}='{v}'".format(k=k, v=v) for (k, v) in env.items()])
    return 'env {variables} sh -c "set -x; {cmd}"'.format(variables=variables, cmd=cmd)

env_common = read_yaml('./env-common.yaml')
env_secret = read_yaml('./env.yaml')
env_combined = dict(env_common, **env_secret)

cmd_raw = "printenv" # this is just for test purposes, I'm using a proper binary in my case
cmd = with_env(cmd_raw, env_combined)
local_resource('my_cmd', cmd)
```

What I would like to do is to avoid this monstrosity and do the following:
```python
env_common = read_yaml('./env-common.yaml')
env_secret = read_yaml('./env.yaml')

cmd = "printenv" # this is just for test purposes, I'm using a proper binary in my case
local_resource('my_cmd', cmd, env=dict(env_common, **env_secret))
```

**Comment by milas** [comment]
This is available with Tilt [v0.18.5+](https://github.com/tilt-dev/tilt/releases/tag/v0.18.5) 🎉 

There's an example in the Local Resource docs in the `serve_cmd` section titled "custom environment": https://docs.tilt.dev/local_resource.html#serve_cmd.

Let us know if you encounter any issues!

**Comment by KarolisL** [comment]
Awesome. Thank you! 🥳
